src/mlpack/core/optimizers/cmaes/cmaes.hpp

```diff
@@ -0,0 +1,177 @@
+/**
+ * @file cmaes.hpp
+ * @author Kartik Nighania (GSoC 17 mentor Marcus Edel)
+ *
+ * Covariance Matrix Adaptation Evolution Strategy
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+#define MLPACK_CORE_OPTIMIZERS_CMAES_CMAES_HPP
+
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace optimization {
+
+class CMAES
+{
+ public:
+CMAES(int objectDim, arma::mat& start, arma::mat& stdDivs,
+double iters = -1.0, double evalDiff = 1e-14);
+
+int getSampleSize(void){ return lambda;}
```

Comment:
Do you mind to follow the style of the rest of the codebase for the parameter functions? e.g.:

```
//! Get the sample size.
size_t const& SampleSize() const { return lambda; }
//! Modify the sample size.
size_t& SampleSize() { return lambda; }
```

```
//! Get the diagonal covariance matrix
arma::mat const& DiagonalCovariance() const { return C.diag(); }
//! Modify the diagonal covariance matrix.
arma::mat& DiagonalCovariance() { return C.diag(); }
```

to set the diagonal covariance matrix we set it with:

```
CMAES optimizer;
optimizer.DiagonalCovariance() = arr;
```